cmd/exchange_wrapper_issues/main.go

```diff
@@ -0,0 +1,821 @@
+
+func parseCLFlags() {
+	exchangesFlag := flag.String("exchanges", "", "A + delimited list of exchange names to run tests against eg -exchanges=bitfinex+anx")
+	assetTypeFlag := flag.String("asset", "", "The asset type to run tests against (where applicable)")
+	currencyPairFlag := flag.String("currency", "", "The currency to run tests against (where applicable)")
+	outputFlag := flag.String("output", "HTML", "JSON, HTML or Console")
+	authenticatedOnlyFlag := flag.Bool("authonly", false, "Skip any wrapper function that doesn't require auth")
+	orderSideFlag := flag.String("orderSide", "BUY", "The order type for all order based wrapper tests")
+	orderTypeFlag := flag.String("orderType", "LIMIT", "The order type for all order based wrapper tests")
+	orderAmountFlag := flag.Float64("orderAmount", 0, "The order amount for all order based wrapper tests")
+	orderPriceFlag := flag.Float64("orderPrice", 0, "The order price for all order based wrapper tests")
+	withdrawAddressFlag := flag.String("withdrawWallet", "", "Withdraw wallet address")
+	flag.Parse()
+	if *exchangesFlag != "" {
+		exchangesToUse = strings.Split(*exchangesFlag, "+")
+	}
+	currencyPairOverride = *currencyPairFlag
+	assetTypeOverride = *assetTypeFlag
+	orderTypeOverride = strings.ToUpper(*orderTypeFlag)
+	orderSideOverride = strings.ToUpper(*orderSideFlag)
+	authenticatedOnly = *authenticatedOnlyFlag
+	output = *outputFlag
+	orderPriceOverride = *orderPriceFlag
+	orderAmountOverride = *orderAmountFlag
+	withdrawAddressOverride = *withdrawAddressFlag
+}
+
+func main() {
+	log.Printf("Loading flags..")
+	parseCLFlags()
+	var err error
+	log.Printf("Loading engine...")
+	engine.Bot, err = engine.New()
+	if err != nil {
+		log.Fatalf("Failed to initialise engine. Err: %s", err)
+	}
+
+	engine.Bot.Settings = engine.Settings{
+		DisableExchangeAutoPairUpdates: true,
+		Verbose:                        false,
+	}
+
+	log.Printf("Loading exchanges..")
+
+	var wg sync.WaitGroup
+	for x := range exchange.Exchanges {
+		name := exchange.Exchanges[x]
+		err = engine.LoadExchange(name, true, &wg)
+		if err != nil {
+			log.Printf("Failed to load exchange %s. Err: %s", name, err)
+			continue
+		}
+	}
+	wg.Wait()
+
+	log.Println("Done.")
+	log.Printf("Testing exchange wrappers..")
+
+	wg = sync.WaitGroup{}
+	var exchangeResponses []ExchangeResponses
+	log.Printf("Loading config...")
+	config, err := loadConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if withdrawAddressOverride != "" {
+		config.WalletAddress = withdrawAddressOverride
+	}
+	if orderTypeOverride != "LIMIT" {
+		config.OrderSubmission.OrderType = orderTypeOverride
+	}
+	if orderSideOverride != "BUY" {
+		config.OrderSubmission.OrderSide = orderSideOverride
+	}
+	if orderPriceOverride > 0 {
+		config.OrderSubmission.Price = orderPriceOverride
+	}
+	if orderAmountOverride > 0 {
+		config.OrderSubmission.Amount = orderAmountOverride
+	}
+
+	for x := range engine.Bot.Exchanges {
+		if len(exchangesToUse) > 0 {
```

Comment:
Could probably do a check for == zero above and use exchangesToUse and deepequal to load only those specific exchanges.